Source/Private/PerfGrapherCommandlet.cpp

```diff
@@ -0,0 +1,188 @@
+{
+    UE_LOG( LogPerfGrapher, Log, TEXT( "--------------------------------------------------------------------------------------------" ) );
+    UE_LOG( LogPerfGrapher, Log, TEXT( "Running MapMetricsGeneration Commandlet" ) );
+
+    TMap< FString, FString > params_map;
+    FMetricsParams metrics_params;
+
+    if ( !ParseParams( params, metrics_params, params_map ) )
+    {
+        UE_LOG( LogPerfGrapher, Error, TEXT( "Failed to parse parameters" ) );
+        return 1;
+    }
+
+    TArray< FString > package_names;
+
+    for ( const auto & param_key_pair : params_map )
+    {
+        if ( param_key_pair.Key == "Map" )
+        {
+            auto map_parameter_value = param_key_pair.Value;
+
+            const auto add_package = [ &package_names ]( const FString & package_name ) {
+                FString map_file;
+                FPackageName::SearchForPackageOnDisk( package_name, nullptr, &map_file );
+
+                if ( map_file.IsEmpty() )
+                {
+                    UE_LOG( LogPerfGrapher, Error, TEXT( "Could not find package %s" ), *package_name );
+                }
+                else
+                {
+                    package_names.Add( *map_file );
+                }
+            };
+
+            add_package( map_parameter_value );
+        }
+    }
+
+    if ( package_names.Num() == 0 )
+    {
+        UE_LOG( LogPerfGrapher, Error, TEXT( "No maps were checked" ) );
+        return 2;
+    }
+
+    for ( const auto & package_name : package_names )
+    {
+        auto * package = LoadPackage( nullptr, *package_name, 0 );
+        if ( package == nullptr )
+        {
+            UE_LOG( LogPerfGrapher, Error, TEXT( "Could not load package %s" ), *package_name );
+            return 2;
+        }
+
+        auto * world = UWorld::FindWorldInPackage( package );
+        if ( world == nullptr )
+        {
+            UE_LOG( LogPerfGrapher, Error, TEXT( "Could not get a world in the package %s" ), *package_name );
+            return 2;
+        }
+
+        world->WorldType = EWorldType::Editor;
+    }
+    return 0;
+}
+
+bool UPerfGrapherCommandlet::ParseParams( const FString & params, FMetricsParams & out_params, TMap< FString, FString > & params_map ) const
+{
+    TArray< FString > tokens;
+    TArray< FString > switches;
+    ParseCommandLine( *params, tokens, switches, params_map );
+
+    // Log initial parameters
+    UE_LOG( LogPerfGrapher, Display, TEXT( "Parsing commandlet parameters:" ) );
+    UE_LOG( LogPerfGrapher, Display, TEXT( "Raw params: %s" ), *params );
+    UE_LOG( LogPerfGrapher, Display, TEXT( "Found %d tokens, %d switches, %d params" ), tokens.Num(), switches.Num(), params_map.Num() );
+
+    for ( const auto & param : params_map )
+    {
+        UE_LOG( LogPerfGrapher, Display, TEXT( "In Param: %s = %s" ), *param.Key, *param.Value );
+    }
+
+    // Set defaults before parsing overrides
+    out_params.MapName = TEXT( "L_OW" );
```

Comment:
no project specific map in this plugin

Reply:
Yeah I know, I wanted to keep this for now to allow for more practical testing. I will remove it.

Reply:
replaced it with a note and default_map. To replace with our map without pushing it.

Reply:
leave it to empty

Reply:
please merge, will do in future PR

Reply:
Should be fine.